android/src/org/ppsspp/ppsspp/PpssppActivity.java

```diff
@@ -135,39 +139,70 @@ public int openContentUri(String uriString, String mode) {
 		}
 	}
 
-	private static String fileInfoToString(DocumentFile file) {
+	private static final String[] columns = new String[] {
+		DocumentsContract.Document.COLUMN_DISPLAY_NAME,
+		DocumentsContract.Document.COLUMN_SIZE,
+		DocumentsContract.Document.COLUMN_FLAGS,
+		DocumentsContract.Document.COLUMN_MIME_TYPE,  // check for MIME_TYPE_DIR
+		DocumentsContract.Document.COLUMN_LAST_MODIFIED
+	};
+
+	private String cursorToString(Cursor c) {
+		final int flags = c.getInt(2);
+		// Filter out any virtual or partial nonsense.
+		// There's a bunch of potentially-interesting flags here btw,
+		// to figure out how to set access flags better, etc.
```

Comment:
FLAG_DIR_SUPPORTS_CREATE/FLAG_SUPPORTS_WRITE could be used for writable, I suppose.

-[Unknown]

Reply:
Indeed, just haven't gotten around to it.